test/test-helpers/pact/pact-interaction-builder.js

```diff
@@ -88,6 +93,10 @@ class PactInteractionBuilder {
       pact.willRespondWith.headers = this.responseHeaders
     }
 
+    if (this.withoutHeaders) {
+      delete pact.willRespondWith.headers
```

Comment:
Connector is returning `Response.noContent().build()`. Our `PactInteractionBuilder` is hard coding `headers: { 'Content-Type': 'application/json' }` to every request/response. Instead of refactoring everything I introduced a new function.